### PR TITLE
Us321046:As a user, in the Jenkins plugin, I can use the execution token instead of username-password for authentication in existing build steps

### PR DESCRIPTION
**Pull request by motteww**
Please Make sure these boxes are checked before submitting your pull request - Thanks ahead!

- [x] Proper pull request title - concise and clear for others.
- [x] Proper pull request short description - clear and concise (as it should appear in the Jira ticket) for other developers and users.
- [x] Proper Jira ticket - Number, Link in pull request description.
- [x] The PR can is merged on your machine without any conflicts.
- [x] The PR can is built on your machine without any (new) warnings.
- [x] The PR passed sanity tests by you / QA / DevTest / Good Samaritan.
- [x] Add unit tests with new features.
- [x] If you added any dependency to the POM - Please update grount

### CODE REVIEW

**Review by dorin7bogdan — COMMENTED** [review]
Hi motteww, please let us know how can we contact you inside MicroFocus. 
Thank you.
Dorin Bogdan.

**Comment by motteww** [comment]
> 

Email or teams to: ying.wang17@microfocus.com

**Comment by Zsolt-Lukacs22** [comment]
Currently unreviewed, please do not merge, someone will handle it next week.

File: HpToolsLauncher/TestRunners/GuiTestRunner.cs [hunk 1]
```diff
@@ -447,7 +447,7 @@ private GuiTestRunResult ExecuteQTPRun(TestRunResults testResults)
                 //Check for cancel before executing
                 if (_runCancelled())
                 {
-                    QTPTestCleanup();
```
Comment (Zsolt-Lukacs22) [inline HpToolsLauncher/TestRunners/GuiTestRunner.cs:450]: Please, discard all changes in GuiTestRunner, unnecessary and interfere with other modifications (causes merge conflict in other PR).

File: src/main/java/com/microfocus/application/automation/tools/mc/HttpUtils.java [hunk 2]
```diff
@@ -176,7 +179,11 @@ private static JSONObject convertStreamToJSONObject(InputStream inputStream) {
                 while ((line = reader.readLine()) != null) {
                     res.append(line);
                 }
-                obj = (JSONObject) JSONValue.parseStrict(res.toString());
+                if (JSONValue.isValidJson(res.toString())) {
+                    obj = (JSONObject) JSONValue.parseStrict(res.toString());
```
Comment (Zsolt-Lukacs22) [inline src/main/java/com/microfocus/application/automation/tools/mc/HttpUtils.java:183]: parseStrict() is unnecessary if isValidJson is called on, can only use parse(), or continue using parseStrict() without additional check, but catching more specific exception types, e.g. parseStrict() can throw ParseException, the IO operations can throw IOException.

File: src/main/java/com/microfocus/application/automation/tools/mc/Oauth2TokenUtil.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            if (s.trim().isEmpty()) {
+                continue;
+            }
+            if (!extractField(s.trim())) {
+                return false;
+            }
+        }
+
+        return true;
```
Comment (Zsolt-Lukacs22) [inline src/main/java/com/microfocus/application/automation/tools/mc/Oauth2TokenUtil.java:26]: I would also add an additional check if all the 3 parts are populated: client, secret, tenant. By default AFAIK, the generated execution token contains all the parts, then if the given token does not contain it is malformed in my opinion.

File: src/main/resources/com/microfocus/application/automation/tools/run/RunFromFileBuilder/config.jelly [hunk 4]
```diff
@@ -251,15 +251,28 @@
...
-					 <f:textbox name="runfromfs.fsUserName" value="${instance.runFromFileModel.fsUserName}"/>
-				 </f:entry>
-				 <f:entry title="Password" field="fsPassword">
-					 <f:password name="runfromfs.fsPassword" value="${instance.runFromFileModel.fsPassword}"/>
-				 </f:entry>
-				 <f:entry title="Tenant Id" field="mcTenantId">
-					 <f:textbox name="runfromfs.mcTenantId" value="${instance.runFromFileModel.mcTenantId}"/>
-				 </f:entry>
+                <f:entry>
+                    <f:radioBlock name="authModel" field="authModel" value="token" checked="${instance.runFromFileModel.authType == 'token' || instance.runFromFileModel.authType == null}" title="Execution Token Authentication">
```
Comment (Zsolt-Lukacs22) [inline src/main/resources/com/microfocus/application/automation/tools/run/RunFromFileBuilder/config.jelly:255]: Would introduce a new separate jelly model for this, as an example SpecifyParametersModel in the new PR, or UFTSettingsModel previously. The jelly modifications are duplicated, a new model would resolve this. If you plan to integrate this, please also introduce a separate JS file for the model, extracting all the code from the configure.js, regarding MC.

**Review by Zsolt-Lukacs22 — CHANGES_REQUESTED** [review]
Please, also plan to resolve most of the CodeClimate analyzer issues and reduce them, as reasonably possible and were introduced by you, previously existing ones are unimportant, but a fix is welcomed.

**Pull request closed without merging**